### PR TITLE
(test) O3-2222: Add tests for `person-attribute-field.component.tsx` component

### PR DESCRIPTION
**Pull request by ayush-AI**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

In this PR I added the tests for person-attribute-field.component.tsx component.

## Screenshots

*None.*

## Related Issue

https://issues.openmrs.org/browse/O3-2222

## Other

*None.*

### CODE REVIEW

**Comment by ayush-AI** [comment]
cc/ @Piumal1999 @anjula-sack @jayasanka-sack

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+jest.mock('../field.resource'); // Mock the useConceptAnswers hook
+
+jest.mock('formik', () => ({
+  ...jest.requireActual('formik'),
+}));
+
+const mockedUsePersonAttributeType = usePersonAttributeType as jest.Mock;
+const mockedUseConceptAnswers = useConceptAnswers as jest.Mock;
+
+let fieldDefinition;
```
Comment (anjula-sack) [inline packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx:18]: Shall we define the type for this?

**Review by anjula-sack — DISMISSED** [review]

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx (hunk 1 above)
Comment (ayush-AI) [reply packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx:18]: for all three or the just the fieldDefinition?

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+const mockedUsePersonAttributeType = usePersonAttributeType as jest.Mock;
+const mockedUseConceptAnswers = useConceptAnswers as jest.Mock;
+
+let fieldDefinition: FieldDefinition;
+
+describe('PersonAttributeField', () => {
+  let mockPersonAttributeType = {
+    format: 'java.lang.String',
+    display: 'Reffered by',
```
Comment (Piumal1999) [inline packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx:24]:
```suggestion
    display: 'Referred by',
```
Update the spellings in other places as well

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      <Formik initialValues={{}} onSubmit={() => {}}>
+        <Form>
+          <PersonAttributeField fieldDefinition={fieldDefinition} />
+        </Form>
+      </Formik>,
+    );
+
+    const input = screen.getByLabelText(/Reffered by/i) as HTMLInputElement;
+    expect(input).toBeInTheDocument();
+    expect(input.type).toBe('select-one');
```
Comment (Piumal1999) [inline packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx:115]: Check for the options (answers) as well

**Review by Piumal1999 — CHANGES_REQUESTED** [review]

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx (hunk 3 above)
Comment (ayush-AI) [reply packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx:115]: done @Piumal1999

File: packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx (hunk 2 above)
Comment (ayush-AI) [reply packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/person-attribute-field.test.tsx]: done @Piumal1999

**Review by Piumal1999 — APPROVED** [review]

**Pull request merged**